### PR TITLE
:sparkles: refactor setting the helm default variable of hostingClusterCapabilities

### PR DESCRIPTION
**Pull request by zhujian7**
## Related issue(s)

Fixes #https://github.com/open-cluster-management-io/addon-framework/pull/246#discussion_r1525070990

### CODE REVIEW

**Review by mprahl — APPROVED** [review]
This looks good to me!

File: pkg/addonfactory/helm_agentaddon.go [hunk 1]
```diff
@@ -258,6 +266,17 @@ func (a *HelmAgentAddon) getDefaultValues(
 
 	if a.hostingCluster != nil {
 		defaultValues.HostingClusterCapabilities = *a.capabilities(a.hostingCluster, addon)
+	} else if a.clusterClient != nil {
+		hostingClusterName := addon.GetAnnotations()[addonapiv1alpha1.HostingClusterNameAnnotationKey]
+		if len(hostingClusterName) > 0 {
+			hostingCluster, err := a.clusterClient.ClusterV1().ManagedClusters().
+				Get(context.TODO(), hostingClusterName, metav1.GetOptions{})
+			if err != nil {
+				klog.Errorf("failed to get hostingCluster %s. err:%v", hostingClusterName, err)
```
Comment (zhiweiyin318) [inline pkg/addonfactory/helm_agentaddon.go:281]: should we ignore err if the hostingCluster is NotFound here ?

Comment (zhujian7) [reply pkg/addonfactory/helm_agentaddon.go:281]: PTAL

**Review by mprahl — APPROVED** [review]

**Comment by mprahl** [comment]
@zhujian7 can this be merged?

**Pull request merged**